pkg/k8sclient/bastions.go

```diff
@@ -0,0 +1,114 @@
+package k8sclient
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/giantswarm/microerror"
+	capg "sigs.k8s.io/cluster-api-provider-gcp/api/v1beta1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+const BastionLabelKey = "cluster.x-k8s.io/deployment-name"
+
+type Bastions struct {
+	client    client.Client
+	finalizer string
+}
+
+func NewBastions(client client.Client, finalizer string) *Bastions {
+	return &Bastions{
+		client:    client,
+		finalizer: finalizer,
+	}
+}
+
+func (b *Bastions) GetBastionIPList(ctx context.Context, cluster *capg.GCPCluster) ([]string, error) {
+	machineList, err := b.getBastionMachineList(ctx, cluster)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	var bastionPublicIPList []string
+
+	for _, machine := range machineList.Items {
+		// no ip address yet
+		if len(machine.Status.Addresses) == 0 {
+			return nil, errors.New("bastion IP is not yet available")
```

Comment:
Do we want to return here? Do we not want to check the rest of the `machine` in the array? (Especially as we have the check on line 51 anyway.)